crates/common/src/dotrain_order/mod.rs

```diff
@@ -180,27 +259,145 @@ impl DotrainOrder {
             dotrain_yaml.cache.update_remote_tokens(remote_tokens);
         }
 
-        Ok(Self {
+        Ok((dotrain, dotrain_yaml))
+    }
+}
+
+/*
+
+NOTE FOR DEVELOPERS:
```

Comment:
@findolor this doesn't really answer why we can't use the constructor with args

Reply:
we can introduce another function that will help us to construct with args on non wasm targets. So in a way we are going to have two different ways of constructing/initializing.

1. Using the `create_with_args` fn that can be used in the repo without the double fn pattern
2. Using the `new constructor` and `create/initialize` fns in wasm targets for class support

how does that sound? i know @rouzwelt is also looking if we can have class support in js side so we can also wait for his findings

Reply:
let's see what @rouzwelt comes up with as a solution